docs/guide/readme.md

```diff
@@ -0,0 +1,269 @@
+        },
+        '.rotate-90': {
+          transform: 'rotate(90deg)',
+        },
+        '.rotate-180': {
+          transform: 'rotate(180deg)',
+        },
+        '.rotate-270': {
+          transform: 'rotate(270deg)',
+        },
+      }
+
+      addUtilities(newUtilities, ['responsive', 'hover'])
+    }
+  ]
+}
+```
+
+Read the [documentation](https://tailwindcss.com/docs/plugins/#adding-utilities) for more informations on plugins.
+
+### Directives and Functions
+
+#### [@apply](https://tailwindcss.com/docs/functions-and-directives#apply)
+
+Apply is useful for [component extraction](https://tailwindcss.com/docs/extracting-components/) when a pattern repeats itself. For buttons for example, it allows to centralize the properties of a component.
+
+These declarations can be used alongside classic CSS declarations.
+
+```scss
+.btn {
+  @apply font-bold py-2 px-4 rounded;
+}
+.btn-blue {
+  @apply bg-blue-500 text-white;
+}
+.btn-blue:hover {
+  @apply bg-blue-700;
+  transform: translateY(-1px);
+}
+```
+
+It is necessary to use interpolation to add `!important` to styles applied via the directive `@apply`:
+
+```scss
+.btn {
+  @apply font-bold py-2 px-4 rounded #{!important};
+}
+```
+
+The `@apply` directive cannot be used for statements about elements states or elemenrs responsiveness. Example of **what you can't do:**
+
+```scss
+.btn {
+  @apply hover:bg-blue-500;
+  @apply md:inline-block;
+}
+```
+
+#### [@variants](https://tailwindcss.com/docs/functions-and-directives/#variants)
+
+It is possible to generate states utilities for our own utility classes by wrapping them in an `@variants` directive.
+
+```scss
+@variants focus, hover {
+  .rotate-0 {
+    transform: rotate(0deg);
+  }
+  .rotate-90 {
+    transform: rotate(90deg);
+  }
+}
+```
+
+This will generate the following CSS:
+
+```css
+.rotate-0 {
+  transform: rotate(0deg);
+}
+.rotate-90 {
+  transform: rotate(90deg);
+}
+
+.focus\:rotate-0:focus {
+  transform: rotate(0deg);
+}
+.focus\:rotate-90:focus {
+  transform: rotate(90deg);
+}
+
+.hover\:rotate-0:hover {
+  transform: rotate(0deg);
+}
+.hover\:rotate-90:hover {
+  transform: rotate(90deg);
+}
+```
+
+**It is important to keep in mind that state variations will be generated in the order in which they are declared.**
+
+Finally, it is possible to generate [customized variations](https://tailwindcss.com/docs/plugins/#adding-variants) to meet more specific needs.
+
+#### [@responsive](https://tailwindcss.com/docs/functions-and-directives#responsive)
+
+It is possible to generate utility classes using the breakpoints defined in the configuration file with the `@responsive` directive.
+
+```scss
+@responsive {
+  .bg-gradient-brand {
+    background-image: linear-gradient(blue, green);
+  }
+}
+```
+
+This will generate the following CSS:
+
+```css
+.bg-gradient-brand {
+  background-image: linear-gradient(blue, green);
+}
+
+/* ... */
+
+@media (min-width: 640px) {
+  .sm\:bg-gradient-brand {
```

Comment:
Should we put the name we use here ? 
s, m, l, xl, etc

Reply:
@ThomasCtnt Since you wrote this guide, could you adjust the readme with our breakpoints?

Reply:
@titouanmathis Yep i'll adjust it

Reply:
@titouanmathis I pushed the modifications to the feature 👍